scripts/reactor_deployment.py

```diff
@@ -8,7 +8,42 @@
 
 
 # # # # # # # # # # # # Constituent Functions # # # # # # # # # # #
+def capacity_increase(df, base_col, rate, start_year, end_year):
+    """
+    This function takes in an increase rate, and creates a new column in the
+    dataframe populated with the increased capacity.
+
+    Parameters
+    ----------
+    df : pandas DataFrame
+        The dataframe to add the new column to.
+    base_col : str
+        The name of the column to use as the base capacity.
+        This column should contain the net capacity.
+    rate : float
+        The percentage of change in capacity year-to-year
+        (e.g. 1.01 for a 1% increase).
+    start_year : int
+        The year to start the increase.
+    end_year : int
+        The year to end the increase.
 
+    Returns
+    -------
+    df : pandas DataFrame
+        The dataframe with the new column added.
+    """
+    for year in range(start_year, end_year):
+        df.loc[year, f"{base_col} Inc {rate}"] = \
+            df.loc[start_year, base_col] * (rate)**(year - start_year)
+
+    for year in range(df.index[0], start_year):
+        df.loc[year, f"{base_col} Inc {rate}"] = df.loc[year, base_col]
```

Comment:
Just a suggestion... 
```suggestion
increase_capacity = lambda x: df.loc[start_year, base_col]*(rate)**(year-start_year) if x >= start_year else df.loc[year, base_col]

 df.loc[year, f"{base_col} Inc {rate}"] = df.loc[:, base_col].apply(increase_capacity) 
```

Your mileage may vary (i.e., I'm not sure this exact implementation would work). But `.apply` is much faster than a for loop.

Reply:
I'll give it a try! So far I haven't really been trying to optimize performance, but I'm down for anything that works faster. I'm always a little hesitant about lambda functions, because I think they border on not human readable, but I'll see what I can do.